graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -844,10 +843,24 @@ def _rename_fields(self, node: ObjectTypeDefinitionNode) -> ObjectTypeDefinition
             # materialize an empty set, making it seem like there are no-op field renamings even
             # when there aren't.
             self.no_op_field_renamings.setdefault(type_name, set()).update(unused_field_renamings)
+        if len(new_field_nodes) == 0:
+            self.object_types_to_suppress.add(type_name)
         new_type_node = copy(node)
         new_type_node.fields = FrozenList(new_field_nodes)
         return new_type_node
 
+    def leave_union_type_definition(
+        self,
+        node: UnionTypeDefinitionNode,
+        key: Any,
+        parent: Any,
+        path: List[Any],
+        ancestors: List[Any],
+    ) -> None:
+        """Check that each union still has at least one non-suppressed member."""
```

Comment:
nit: This docstring doesn't really capture what's going on. The implementation does an action if that check fails, but the docstring doesn't make any mention of that. 

Consider instead:
```suggestion
        """Suppress unions that do not have any remaining non-suppressed members."""
```

Reply:
The action that happens if the check fails is actually raising an error instead of automatically suppressing the union (since renaming is defined so that unions in such situations have to get explicitly suppressed) and the error handling code is outside of the visitor itself, so I'm inclined to keep the docstring as-is. Worth adding a bit to mention what the `union_types_to_suppress` field is going to be used for, even if it doesn't happen in this function itself?